lib/logstash/inputs/kafka.rb

```diff
@@ -210,6 +210,20 @@ class LogStash::Inputs::Kafka < LogStash::Inputs::Base
   config :security_protocol, :validate => ["PLAINTEXT", "SSL", "SASL_PLAINTEXT", "SASL_SSL"], :default => "PLAINTEXT"
   # SASL client callback handler class
   config :sasl_client_callback_handler_class, :validate => :string
+  # The URL for the OAuth 2.0 issuer token endpoint.
+  config :sasl_oauthbearer_token_endpoint_url, :validate => :string
+  # (optional) The override name of the scope claim.
+  config :sasl_oauthbearer_scope_claim_name, :validate => :string, :default => 'scope'
+  # SASL login callback handler class
+  config :sasl_login_callback_handler_class, :validate => :string
+  # (optional) The duration, in milliseconds, for HTTPS connect timeout
+  config :sasl_login_connect_timeout_ms, :validate => :number, :default => 10000
+  # (optional) The duration, in milliseconds, for HTTPS read timeout.
+  config :sasl_login_read_timeout_ms, :validate => :number, :default => 10000
+  # (optional) The duration, in milliseconds, to wait between HTTPS call attempts.
+  config :sasl_login_retry_backoff_ms, :validate => :number, :default => 100
+  # (optional) The maximum duration, in milliseconds, for HTTPS call attempts.
+  config :sasl_login_retry_backoff_max_ms, :validate => :number, :default => 10000
```

Comment:
Will these settings. have any effect for producers and/or consumers? I can see these defined in [broker configs](https://docs.confluent.io/platform/current/installation/configuration/broker-configs.html#sasl-login-refresh-buffer-seconds), but not for [producers](https://docs.confluent.io/platform/current/installation/configuration/producer-configs.html#sasl-login-callback-handler-class) or [consumers](https://docs.confluent.io/platform/current/installation/configuration/consumer-configs.html#sasl-login-callback-handler-class)?

Reply:
I think so, for example: https://docs.confluent.io/platform/current/installation/configuration/producer-configs.html#sasl-login-retry-backoff-max-ms Maybe i'm missing some nuance though in what you are trying to point out?

Reply:
No, you're right. I'm not quite sure how I missed it. 🤦